src/recorder/jamrecorder.cpp

```diff
@@ -314,30 +314,39 @@ void CJamRecorder::Init( const CServer* server,
         throw std::runtime_error( (recordBaseDir.absolutePath() + " is a directory but cannot be written to").toStdString() );
     }
 
-    QObject::connect((const QObject *)server, SIGNAL ( Stopped() ),
-                     this, SLOT( OnEnd() ),
-                     Qt::ConnectionType::QueuedConnection);
+    QObject::connect( (const QObject *)server, SIGNAL ( Stopped() ),
+                      this, SLOT( OnEnd() ),
+                      Qt::ConnectionType::QueuedConnection );
```

Comment:
I was wondering why you are using a "QueuedConnection" for your connections. Is this really required?

Reply:
Possibly not here as it's the shut down.  But generally, the idea was that the main server thread just queues the request and the recorder thread reads the queue when it "gets around to it".  My understanding is that this is the most asynchronous approach to the signal/slot connections.  I could be misunderstanding it, of course! :)

Reply:
Since you created a speparate thread for the recorder, the connections should automtically be a queued connection. But it does not hurt to set it explicitly here. Thanks for the clarification.